components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -3354,6 +3427,44 @@ def listOrphanedAnnotations(self, parent_type, parent_ids, eid=None,
         for e in q.findAllByQuery(sql, p, self.SERVICE_OPTS):
             yield AnnotationWrapper._wrap(self, e)
 
+    def getAnnotationCounts(self, objDict=None):
+        """
+        Get the annotion counts for the given objects
+        """
+        obj_type = None
+        obj_ids = []
+        if objDict is not None:
+            for key in objDict:
+                for o in objDict[key]:
+                    if obj_type is not None and obj_type != key:
+                        raise AttributeError(
+                            "getAnnotationCounts cannot be used with "
+                            "different types of objects")
+                    obj_type = key
+                    obj_ids.append(o.id)
+
+        if obj_type is None:
+            return countAnnotations()
+
+        obj_type = obj_type.title().replace("Plateacquisition",
+        "PlateAcquisition")
+
+        params = omero.sys.ParametersI()
+        params.addIds(obj_ids)
+        q = """
+            select al from %sAnnotationLink al
+            left outer join al.parent as pa
```

Comment:
I think if you don't use parent you don't need this line , it is enough `where al.parent in (:ids)`